### ISSUE TITLE
Test accessibility

### ISSUE THREAD

**Issue by nacnudus** [ukgovdatascience/govdown#11]

**Comment by nacnudus** [comment]
Inherited from https://github.com/ukgovdatascience/rap_companion/issues/83 and related to the [RAP website](https://ukgovdatascience.github.io/rap-website/companion-index.html).

**Comment by nacnudus** [comment]
Notes from using the [personas](https://accessibility.blog.gov.uk/2019/02/11/using-persona-profiles-to-test-accessibility/) set up in the GDS empathy lab.

## Claudia

### Access needs

* 54 years old social worker
* Sight impaired
* Uses a screen magnifier and changes colours to increase contrast

### Notes

* Page loads in the middle at “work with data using methods adapted from software…”  So does GOV.UK
* Link to RAP Champions network page (GSS site) does load all text.  Stops below “restricted to the production of official statistics, it is open to anyone in government”.  Can only scroll past with arrow keys.  Similar think can happen on RAP Website and GOV.UK if the arrow keys have already been scrolled.
* Link to GitHub issues works, destination seems okay.
* It’s a pain to scroll across the page.  Ditto on GOV.UK.  Can’t it be made to use the Mobile layout at that magnification?
* Original RAP bookdown website page-scroll links are way off to the side in the middle of nowhere at high zoom.  When you click them, they open the next page in the middle of nowhere.
* Graphs have data points far from the axes.
* Original RAP bookdown website right and left arrow keys scroll the pages, which is unexpected.
* Tables don’t have row headers.
* Original RAP bookdown website you have to tab through the entire table of contents before getting to the actual content.

## Simone

### Access needs

* 41 years old office manager
* Dyslexic
* Uses software which reads out and highlights text

### Notes

* Code chunks aren’t read verbatim.  E.g. `git push origin feature/post_name -u` doesn’t read the dashes aloud.
* Bulleted lists are read out but not described as bulleted.
* “GitLab” is pronounced “DgitLab” with a soft G but “GitHub” is pronounced with a hard G.
* Reader isn’t good at recognising acronyms to be spelled letter by letter.
* “10-11am” is read “ten eleven a m”.
* D3 and RShiny are read as expected

## Pawel

### Access needs

* 24 years old chemistry graduate
* Asperger’s and anxiety
* Adjusts things, like colours, to reduce potential stress and distractions

### Notes

* Images are incredibly distracting, including graphs.
* Text is also incredibly distracting -- it flashes.
* Long paragraphs are hard to read because they flash and when they reappear it is hard to find your place.
* Adjusted colours work well.  Code isn’t invisible.

## Ashleigh

### Access needs

* 24 years old arts graduate and administrative assistant
* Severely sight impaired (blind)
* Uses a screen reader

### Notes

* Tabbing through the page for the screen reader doesn’t navigate by heading.  GOV.UK is the same.

## Ron

### Access needs

* 82 years old, retired
* Multiple conditions: arthritis, losing his hearing, cataracts, hip replacements
* Doesn’t use any assistive technology

### Notes

* Links in text are very difficult to click -- poor precision with the mouse, doesn’t use the keyboard.

## Chris

### Access needs

* 53 y/o management accountant
* Rheumatoid arthritis
* Uses keyboard only, no mouse and just started using voice control

### Notes

* Couldn’t get voice control to work in the lab (saying ‘wake up’ didn’t activate it, nor did restarting the extension).

## Saleem

### Access needs

* 22 years old, unemployed
* Deaf
* BSL first, English second language, always has video captions switched on.

### Notes

* No videos on the site.
* Technical language is very difficult to read in Pig Latin.

**Comment by nacnudus** [comment]
Guidance for non-specialist assessment here: https://accessibility-self-evaluation.london.cloudapps.digital/#doing-a-basic-accessibility-check-if-you-can-t-do-a-detailed-one (contact me for login)

**Comment by nacnudus** [comment]
I've been through the site again checking against [WCAG 2.1 Level AA](https://www.w3.org/WAI/WCAG21/quickref/).

There was a problem with invisible focus in code blocks, which has been fixed by disabling code highlighting (see https://github.com/ukgovdatascience/govdown/issues/45).

There was also only one link to each page, so now each page is also linked to from the first section of the home page.
